### PR TITLE
[#5204] feat(iceberg): support  more  table event for Iceberg REST server

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

support more table event for Iceberg REST server

### Why are the changes needed?

Fix: #5204 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao , please help to review when you have time, thanks

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java [hunk 1]
```diff
@@ -144,19 +149,20 @@ public Response updateTable(
       @PathParam("namespace") String namespace,
       @PathParam("table") String table,
       UpdateTableRequest updateTableRequest) {
+    String catalogName = IcebergRestUtils.getCatalogName(prefix);
+    Namespace icebergNS = RESTUtil.decodeNamespace(namespace);
     if (LOG.isInfoEnabled()) {
       LOG.info(
-          "Update Iceberg table, namespace: {}, table: {}, updateTableRequest: {}",
-          namespace,
+          "Update Iceberg table, catalog:{}, namespace: {}, table: {}, updateTableRequest: {}",
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java:156]: Add a space after "catalog:"

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java [hunk 2]
```diff
@@ -169,18 +175,16 @@ public Response dropTable(
       @PathParam("namespace") String namespace,
       @PathParam("table") String table,
       @DefaultValue("false") @QueryParam("purgeRequested") boolean purgeRequested) {
+    String catalogName = IcebergRestUtils.getCatalogName(prefix);
+    Namespace icebergNS = RESTUtil.decodeNamespace(namespace);
     LOG.info(
-        "Drop Iceberg table, namespace: {}, table: {}, purgeRequested: {}",
-        namespace,
+        "Drop Iceberg table, catalog:{}, namespace: {}, table: {}, purgeRequested: {}",
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java:181]: Also here.

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableRenameOperations.java [hunk 3]
```diff
@@ -57,7 +60,13 @@ public IcebergTableRenameOperations(IcebergCatalogWrapperManager icebergCatalogW
   @ResponseMetered(name = "rename-table", absolute = true)
   public Response renameTable(
       @PathParam("prefix") String prefix, RenameTableRequest renameTableRequest) {
-    icebergCatalogWrapperManager.getOps(prefix).renameTable(renameTableRequest);
+    String catalogName = IcebergRestUtils.getCatalogName(prefix);
+    LOG.info(
+        "List Iceberg tables, catalog:{}, source: {}, destination: {}.",
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableRenameOperations.java:65]: Is it "list" or "rename" operation?

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRenameTableEvent.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.gravitino.iceberg.service.IcebergRestUtils;
+import org.apache.iceberg.rest.requests.RenameTableRequest;
+
+/** Represent an event after rename Iceberg table successfully. */
+@DeveloperApi
+public class IcebergRenameTableEvent extends IcebergTableEvent {
+  private RenameTableRequest renameTableRequest;
+
+  public IcebergRenameTableEvent(
+      String user, NameIdentifier resourceIdentifier, RenameTableRequest renameTableRequest) {
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRenameTableEvent.java:33]: Should here be the response or result after the table is renamed?

Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRenameTableEvent.java:33]: There is no response if rename table successfully.  The Iceberg REST API Spec:
```
      responses:
        204:
          description: Success, no content
        400:
          $ref: '#/components/responses/BadRequestErrorResponse'
        401:
          $ref: '#/components/responses/UnauthorizedResponse'
        403:
          $ref: '#/components/responses/ForbiddenResponse'
        404:
          description:
            Not Found
            - NoSuchTableException, Table to rename does not exist
            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
```

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java (hunk 1 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java]: updated

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java [hunk 5]
```diff
@@ -101,8 +102,12 @@ public IcebergTableOperations(
   @ResponseMetered(name = "list-table", absolute = true)
   public Response listTable(
       @PathParam("prefix") String prefix, @PathParam("namespace") String namespace) {
-    return IcebergRestUtils.ok(
-        icebergCatalogWrapperManager.getOps(prefix).listTable(RESTUtil.decodeNamespace(namespace)));
+    String catalogName = IcebergRestUtils.getCatalogName(prefix);
+    Namespace icebergNS = RESTUtil.decodeNamespace(namespace);
+    LOG.info("List Iceberg tables, catalog:{}, namespace: {}", catalogName, icebergNS);
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java:107]: Can you please add a whitespace after ":"?

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java [hunk 6]
```diff
@@ -195,21 +199,23 @@ public Response loadTable(
       @PathParam("table") String table,
       @DefaultValue("all") @QueryParam("snapshots") String snapshots,
       @HeaderParam(X_ICEBERG_ACCESS_DELEGATION) String accessDelegation) {
+    String catalogName = IcebergRestUtils.getCatalogName(prefix);
+    Namespace icebergNS = RESTUtil.decodeNamespace(namespace);
     boolean isCredentialVending = isCredentialVending(accessDelegation);
     LOG.info(
-        "Load iceberg table, namespace: {}, table: {}, access delegation: {}, "
+        "Load iceberg table, catalog: {}, namespace: {}, table: {}, access delegation: {}, "
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java:206]: Can you please change all the "iceberg" to "Iceberg" here and below?

**Comment by jerryshao** [comment]
Just two minor issues, others LGTM.

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java (hunk 5 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java]: updated

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java (hunk 6 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/rest/IcebergTableOperations.java]: updated

**Review by jerryshao — APPROVED** [review]

**Pull request merged**